subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -191,6 +202,82 @@ public void resolveArtifacts(ConfigurationInternal configuration, ResolverResult
         results.artifactsResolved(new DefaultResolvedConfiguration(result), result);
     }
 
+    @Override
+    public List<ArtifactRepository> getRepositories() {
+        List<ExposableRepository> exposableRepositories = CollectionUtils.collect(repositories, Transformers.cast(ExposableRepository.class));
+        List<ArtifactRepository> result = Lists.newArrayListWithExpectedSize(exposableRepositories.size());
+        for (ExposableRepository repository : exposableRepositories) {
+            ConfiguredModuleComponentRepository resolver = repository.createResolver();
```

Comment:
I think this means we are going to create an extra resolver now. Can we reuse?

Reply:
:x: I don't like that we create a resolver here. You don't need a resolver, you need information about the repository. The fact that we're using a resolver to get this information is accidental. I think the information you need should be exposed in another way.